test/unit/deployment_test.rb

```diff
@@ -13,4 +13,61 @@ class DeploymentTest < ActiveSupport::TestCase
       assert_equal previous, the_deploy.previous_deployment
     end
   end
+
+  describe "#commit_match?" do
+    should "return true when SHAs are the same" do
+      commit_sha = "c579613e5f0335ecf409fed881fa7919c150c1af"
+      deployment = FactoryBot.create(
+        :deployment,
+        application: FactoryBot.create(:application),
+        deployed_sha: commit_sha,
+      )
+
+      assert_equal true, deployment.commit_match?(commit_sha)
+    end
+
+    should "return false when SHAs are the different" do
+      commit_sha = "c579613e5f0335ecf409fed881fa7919c150c1af"
+      deployment = FactoryBot.create(
+        :deployment,
+        application: FactoryBot.create(:application),
+        deployed_sha: "0fa90a3bc5b1c91e9355de3507244e11d8a2d68c",
+      )
+
+      assert_equal false, deployment.commit_match?(commit_sha)
+    end
+
+    should "return true if deployed_sha is a short SHA" do
+      commit_sha = "c579613e5f0335ecf409fed881fa7919c150c1af"
+      deployment = FactoryBot.create(
+        :deployment,
+        application: FactoryBot.create(:application),
+        deployed_sha: commit_sha[...7],
```

Comment:
optional nit: I'd write string literals here as well, so that it's clear at a glance what the assertion is and there's no possibility of off-by-ones etc.